spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/util/ResourceExtractor.java

```diff
@@ -0,0 +1,158 @@
+ * ]
+ * </code>
+ *
+ * Note that all resources but the docker-compose-debug-dataflow.yml are copied to the temp folder.
+ *
+ * @author Christian Tzolov
+ */
+public class ResourceExtractor {
+
+	private final Logger logger = LoggerFactory.getLogger(ResourceExtractor.class);
+
+	/**
+	 * Temporal folder where all resources are extracted.
+	 */
+	private final Path rootPath;
+
+	/**
+	 * Spring utility for loading resources (e.. class path or file system resources)
+	 */
+	private final ResourceLoader resourceLoader;
+
+	public ResourceExtractor(Path rootPath) {
+		this.rootPath = rootPath;
+		this.resourceLoader = new DefaultResourceLoader();
+	}
+
+	/**
+	 * From a list of input resource URIs, copies the resources content under the {@link #rootPath} folder
+	 * using the file name provided with the input URIs!
+	 *
+	 * If ann input resource URI doesn't specify an explicit schema prefix (e.g. classpath:, http:, fie:)
+	 * the resource is not extracted to the {@link #rootPath} folder folder but left in place. Also in that case
+	 * the input resource URI returned as it is.
+	 *
+	 * @param resourceUris list of resource URIs to be extracted.
+	 * @return Returns list of the new file URIs after the extraction. If an input URI didn't use
+	 * and explicit URI schema prefix the input URI is returned unmodified.
+	 */
+	public String[] extract(String... resourceUris) {
+		return Arrays.stream(resourceUris)
+				.map(this::extract)
+				.filter(Objects::nonNull)
+				.collect(Collectors.toList())
+				.toArray(new String[resourceUris.length]);
+	}
+
+	/**
+	 * For an input resource URI, such as classpath:/, file:/ or http/https:/, copies the resource content
+	 * into a new file under the {@link #rootPath} folder using the file name provided with the URI!
+	 *
+	 * If the input resource URI doesn't specify an explicit schema prefix (e.g. classpath:, http:, fie:)
+	 * the resource is not extracted but left in place and the input resource URI is returned.
+	 *
+	 * @param resourceUri Resource URI to be extracted. If the URI doesn't specify an explicit schema prefix
+	 *                    the resource is not modified (e.g. it is left in place).
+	 * @return The new URI location of the copied resource. If the input resource URI doesn't use and
+	 * explicit URI schema prefix (e.g. classpath:, http:, fie:) the input URI is returned unmodified.
+	 */
+	public String extract(String resourceUri) {
+		if (!resourceUri.contains(":")) {
+			return resourceUri;
+		}
+
+		try {
+			Resource resource = this.resourceLoader.getResource(resourceUri);
+			Path localResourcePath = Paths.get(this.rootPath.toString(), resource.getFilename());
+			FileCopyUtils.copy(resource.getInputStream(),
+					new FileOutputStream(localResourcePath.toFile()));
+			return localResourcePath.toString();
+		}
+		catch (IOException e) {
+			logger.error("Failed to extract:" + resourceUri, e);
+		}
+		return null;
+	}
+
+	public static void main(String[] args) {
```

Comment:
Should we need this main method?

Reply:
No, i left it just as an example but will remove it